srcalc/src/main/java/gov/va/med/srcalc/db/HistoricalSearchParameters.java

```diff
@@ -0,0 +1,148 @@
+package gov.va.med.srcalc.db;
+
+import java.util.Objects;
+
+import org.hibernate.criterion.DetachedCriteria;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Restrictions;
+import org.joda.time.LocalDate;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Optional;
+
+import gov.va.med.srcalc.domain.calculation.HistoricalCalculation;
+
+/**
+ * <p>Encapsulates search parameters for {@link HistoricalCalculation}s.</p>
+ * 
+ * <p>Note: in most classes, we define the interface very tightly (using Optional objects
+ * instead of nulls, etc.), but here we are pretty flexible in what we accept to allow
+ * Spring to bind forms to these objects.</p>
+ * 
+ * <p>Per Effective Java Item 17, this class is marked final because it was not
+ * designed for inheritance.</p>
+ */
+public final class HistoricalSearchParameters
+{
+    /**
+     * The maximum number of results a search will return.
+     */
+    public static final int MAX_RESULTS = 1000;
+    
+    private Optional<LocalDate> fMinDate = Optional.absent();
+    private Optional<LocalDate> fMaxDate = Optional.absent();
+    
+    /**
+     * Returns the minimum start date to filter the results. In other words, no results
+     * will be returned with a start date before this date. Null means no minimum date
+     * filter. Default: null.
+     * @return nullable
+     */
+    public LocalDate getMinDate()
+    {
+        return fMinDate.orNull();
+    }
+    
+    /**
+     * Sets the minimum start date to filter the results.
+     * @param startDate may be null
+     * @see #getMinDate()
+     */
+    public void setMinDate(final LocalDate startDate)
+    {
+        fMinDate = Optional.fromNullable(startDate);
+    }
+    
+    /**
+     * Returns the maximum start date to filter the results. In other words, no results
+     * will be returned with a start date after this date. Null means no maximum date
+     * filter. Default: null.
+     * @return nullable
+     */
+    public LocalDate getMaxDate()
+    {
+        return fMaxDate.orNull();
+    }
+    
+    /**
+     * Sets the maximum start date to filter the results.
+     * @param endDate may be null
+     * @see #getMaxDate()
+     */
+    public void setMaxDate(final LocalDate endDate)
+    {
+        fMaxDate = Optional.fromNullable(endDate);
+    }
+    
+    /**
+     * Constructs a Hibernate {@link DetachedCriteria} that will perform the search. No
+     * maximum is set (because the {@link DetachedCriteria} interface does not allow it):
+     * that is the calling code's responsibility.
+     * @return a DetachedCriteria on {@link HistoricalCalculation} objects
+     */
+    DetachedCriteria makeCriteria()
+    {
+        final DetachedCriteria criteria =
+                DetachedCriteria.forClass(HistoricalCalculation.class);
+        criteria.addOrder(Order.desc("startTimestamp"));
+        
+        if (fMinDate.isPresent())
+        {
+            criteria.add(Restrictions.ge(
```

Comment:
Nothing wrong here, but the abbreviated method names make me think something is wrong every time. (i.e. ge, lt, le etc.)